kafka/common/pkg/kafka/consumer_factory.go

```diff
@@ -66,44 +65,34 @@ func (c *customConsumerGroup) Errors() <-chan error {
 	return errors
 }
 
-func (c *customConsumerGroup) Close() error {
-	c.closeFn()
-	return c.ConsumerGroup.Close()
-}
-
 var _ sarama.ConsumerGroup = (*customConsumerGroup)(nil)
 
 func (c kafkaConsumerGroupFactoryImpl) StartConsumerGroup(groupID string, topics []string, logger *zap.Logger, handler KafkaConsumerHandler) (sarama.ConsumerGroup, error) {
-	consumerGroup, err := newConsumerGroupFromClient(groupID, c.client)
-
+	consumerGroup, err := c.newConsumerGroupFunc(c.addrs, groupID, c.config)
 	if err != nil {
 		return nil, err
 	}
 
 	consumerHandler := NewConsumerHandler(logger, handler)
 
-	ctx, cancelFn := context.WithCancel(context.Background())
-
 	go func() {
+		ctx := context.Background()
 		for {
-			if err2 := consumerGroup.Consume(context.Background(), topics, &consumerHandler); err2 != nil {
-				consumerHandler.errors <- err2
-			}
-			// Let's check if Consume stopped because of closing
-			select {
-			case <-ctx.Done():
```

Comment:
I would prefer to keep the close propagation, TBH I mistrust how these sarama internals works that are not defined in their public api docs

Reply:
The documentation states 
> ErrClosedConsumerGroup is the error returned when a method is called on a consumer group that has been closed.

Anyway, let's double-check.

Reply:
> The documentation states

What really makes me uncomfortable is that this is the err documentation but not  the `ConsumerGroup.Consume` documentation...

Reply:
sarama is _funny_: if you close the consumer group, it could return nil err more than `ErrConsumerGroupClosed` https://github.com/cloudevents/sdk-go/pull/488/files#diff-0a44b98f7234278f1a60826ec8c22e70R151 (spotted with some debugging :smile: )

Reply:
yes yes we catch it, just to let you know how weirdly it behaves :smile: